tests/test_core.py

```diff
@@ -512,6 +512,23 @@ def test_dagobah_expand_simple_job():
 
 
 @with_setup(blank_dagobah)
+def test_dagobah_expand_same_job_twice():
+    dagobah.add_job('test_job_a')
```

Comment:
Failure conditions for what? This fix is to fix a false negative, ie the job fails when it is supposed to pass, so I tested for the most basic case. I'm not sure how I'd test this issue in an inverse or converse manner, without repeating existing tests.

There are already a few tests to ensure that job expansion fails when it is supposed to fail:

* https://github.com/rclough/dagobah/blob/rmc-JIJ-cycle-fix/tests/test_core.py#L388
* https://github.com/rclough/dagobah/blob/rmc-JIJ-cycle-fix/tests/test_core.py#L402